.github/workflows/pre-commit-actions.yml

```diff
@@ -0,0 +1,37 @@
+name: precommit-actions
+run-name: ${{ github.actor }} is running precommit actions.
+on:
+  push:
+    branches-ignore: [main]
+  pull_request:
+    branches: [main]
+jobs:
+  build:
+      name: Lint
+      runs-on: ubuntu-latest
```

Comment:
I'm not too familiar with github actions, but will this line affect Windows compatibility?

Reply:
Shouldn't it's just base platform to run on which to test `flake8` guidelines.

Reply:
Perfect then. Thanks for answering the question.